### PR TITLE
Add seam unit tests

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

**Comment by andreasohlund** [comment]
@tmasternak the learning transport failures seems to be unrelated to the changes?

File: src/ServiceControl.Transports.ASBS/QueueLengthProvider.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace ServiceControl.Transports.ASBS
     using Microsoft.Azure.ServiceBus.Management;
     using NServiceBus.Logging;
 
-    public class QueueLengthProvider : IProvideQueueLengthNew
+    class QueueLengthProvider : IProvideQueueLengthNew
```
Comment (tmasternak) [inline src/ServiceControl.Transports.ASBS/QueueLengthProvider.cs]: Can we change the name of the `IProvieQueueLengthNew`?

File: src/ServiceControl.Transports.Learning/QueueLengthNewProvider.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@
     using NServiceBus.Logging;
 
 
-    public class QueueLengthNewProvider : IProvideQueueLengthNew
+    class QueueLengthNewProvider : IProvideQueueLengthNew
```
Comment (tmasternak) [inline src/ServiceControl.Transports.Learning/QueueLengthNewProvider.cs]: Can we rename `QueueLengthNewProvider`?

**Comment by andreasohlund** [comment]
I was going to do that in a separate PR?

**Comment by tmasternak** [comment]
Regarding 

> @tmasternak the learning transport failures seems to be unrelated to the changes?

That was a bug that I fixed in the main branch. Waiting for the test to verify.

**Comment by andreasohlund** [comment]
Test failures are unrelated so merging this

**Pull request merged**